rero_mef/agents/viaf/api.py

```diff
@@ -310,27 +438,33 @@ def get_online_record(cls, viaf_source_code, pid, format=None):
                     # get pid
                     text = source.get('#text', '|')
                     text = text.split('|')
-                    if bib_source := cls.sources.get(text[0]):
+                    if bib_source := cls.sources.get(text[0], {}).get('name'):
                         result[f'{bib_source}_pid'] = text[1]
                         # get URL
                         if nsid := source.get('@nsid'):
                             if nsid.startswith('http'):
                                 result[bib_source] = nsid
-                # get Wikipedia and WorldCat URLs
-                if result.get('wiki_pid'):
-                    x_links = data_json.get('xLinks', {}).get('xLink', [])
-                    for x_link in x_links:
-                        if 'worldcat' in x_link:
-                            result['worldcat'] = x_link
-                        elif isinstance(x_link, dict):
-                            text = x_link.get('#text')
-                            if text and 'wikipedia' in text:
-                                result.setdefault('wiki', []).append(text)
+                # get Wikipedia URLs
+                x_links = data_json.get('xLinks', {}).get('xLink', [])
+                if not isinstance(x_links, list):
+                    x_links = [x_links]
+                for x_link in x_links:
+                    if isinstance(x_link, dict) and result.get('wiki_pid'):
```

Comment:
I didn't find where `wiki_pid` key is set. If it's never set, all this `for` statement is useless.
As main remarks, extract the `result.get('wiki_pid')` statement outside of the loop to avoid enter into is you don't have wiki_pid 

```python
if result.get('wiki_pid'):
    for x_link in filter(lambda x: isinstance(x, dict), x_links):
        text = x_link.get('#text')
        ...
```

Reply:
`wiki_pid` is set in this line:
```python
result[f'{bib_source}_pid'] = text[1]
```